consumer/consumer.go

```diff
@@ -315,9 +342,22 @@ func (c *Consumer) channelState(conn AMQPConnection, connCloseCh <-chan struct{}
 			return c.waitRetry(err)
 		}
 
+		err = ch.Qos(c.prefetchCount, 0, c.qosGlobal)
+		if err != nil {
+			return c.waitRetry(err)
+		}
+
 		queue := c.queue
+		if c.queueDeclare || queue == "" {
```

Comment:
`if c.queueDeclare ` should be enough

Reply:
i think we should leave as is `|| queue == ""` because we don’t have other conditions for declaring queue when only WithExchange option passed. 
If we use validation into `New()` instead of reset mutual WithExchange or WithQueue params and it is only way for declaring temporary queue.